src/main/scala/zio/s3/providers.scala

```diff
@@ -42,6 +42,14 @@ object providers {
           .mapError(err => InvalidCredentials(err.getMessage))
       )
 
+  def profile(name: String): ZManaged[Blocking, InvalidCredentials, ProfileCredentialsProvider] =
```

Comment:
Thanks for your contribution. What about this 👇 

```scala
val profile: ZManaged[Blocking, InvalidCredentials, ProfileCredentialsProvider] =
    profile(None)

def profile(name: Option[String]): ZManaged[Blocking, InvalidCredentials, ProfileCredentialsProvider] =
    ZManaged
      .fromAutoCloseable(IO.succeed(ProfileCredentialsProvider.create(name.orNull)))
      .tapM(c =>
        effectBlocking(c.resolveCredentials())
          .mapError(err => InvalidCredentials(err.getMessage))
      )
```

Reply:
I figured it was nicer to call `profile("some profile")` than `profile(Some("some profile"))` but not a strong feeling. will update